app/scripts/modules/core/src/presentation/forms/stories/FormikIntro.stories.mdx

```diff
@@ -0,0 +1,117 @@
+import { ArgsTable, Meta, Story, Canvas } from '@storybook/addon-docs/blocks';
+
+import { FormikFormField } from '../';
+import {URLComponent} from './SpinFormikStoriesHelper'
+
+<Meta
+  title="Forms/Intro"
+  decorators={[
+    (Story) => (
+      <div style={{ background: '#fff', border: '1px solid #eee', borderRadius: 10, padding: 20, width: 500 }}>
+        <Story />
+      </div>
+    ),
+  ]}
+  parameters={{
+    layout: 'centered',
+  }}
+/>
+
+Spinnaker forms are built upon [Formik](https://formik.org) and provides two key abstractions for easily building forms: `SpinFormik` and `FormikFormField`.
+
+## `SpinFormik`
+
+This component is a thin wrapper around [`Formik`](https://formik.org/docs/api/formik) component. A typical form is created the following way
+
+```jsx
+interface FormValues {
+  field1: string;
+  field2: number;
+}
+
+<SpinFormik<FormValues>
+  initialValues={{ field1: 'foo', field2: 2 }}
+  render={(formikProps) => <YourFormView {...formikProps} />}
+  onSubmit={(values) => {
+    /* handle form submission */
+  }}
+/>
+```
+
+### Props
+
+`SpinFormik` passes all props to `Formik`, so check out the [documentation](https://formik.org/docs/api/formik#reference) for detailed information on props accepted by `Formik`.
+
+## `FormikFormField`
+
+This component handles the communication with formik and provides the necessary props to your input fields. Render your fields by passing them to the `input` prop.
+
+```jsx
+<SpinFormik<FormValues>
+  initialValues={{ field1: 'foo', field2: 2 }}
+  render={(formikProps) => (
+    <form onSubmit={formikProps.handleSubmit}>
+      <FormikFormField name="field" label="Field 1" input={(inputProps) => <TextInput {...inputProps} />} />
+    </form>
+  )}
+  onSubmit={(values) => {
+    /* handle form submission */
+  }}
+/>
+```
+
+### Props
+
+<ArgsTable of={FormikFormField} />
+
+Here's a simple form with all supported input fields
+
+<!-- Simple Form Example -->
+<Canvas>
+  <Story id="forms-spinformik--simple-form" />
+</Canvas>
+
+## Validation
+
+Forms can be validated by providing a `validate` function to `FormikFormField`. The validate function is expected to return an error message when the field value is invalid and `FormikFormField` automatically renders the error message.
```

Comment:
We should document when inline `validate` on the `FormikFormField` is acceptable and when a `validate` function should be provided for the overall form.

Again, good for now but let's come back to this later.